File: internal/catalog/pickpool.go
package catalog

import (
	"github.com/lukegabriel520/workit/internal/config"
	"github.com/lukegabriel520/workit/internal/launch"
)

func GetProfilePickPool(profile config.Profile) []PickableItem {
	paths := launch.GetGameLauncherPaths()
	catalog := ResolveCatalogPickables(profile.CatalogGameIDs, paths)

	var custom []PickableItem
	if profile.CustomGamesFolder != "" {
		custom = ScanCustomGamesFolder(profile.CustomGamesFolder)
	}

	seen := map[string]struct{}{}
	var merged []PickableItem

	for _, item := range append(catalog, custom...) {
		if _, ok := seen[item.PickID]; ok {
			continue
		}
		seen[item.PickID] = struct{}{}
		merged = append(merged, item)
	}
	return merged
}

func ProfileHasPickPool(profile config.Profile) bool {
	return len(GetProfilePickPool(profile)) > 0
}

File: src/config/pick-pool.ts
import type { PickableItem } from "./custom-games.js";
import { scanCustomGamesFolder } from "./custom-games.js";
import { resolveCatalogPickables } from "./games.js";
import type { Profile } from "./schema.js";

export function getProfilePickPool(profile: Profile): PickableItem[] {
  const catalog = resolveCatalogPickables(profile.catalogGameIds ?? []);
  const custom = profile.customGamesFolder
    ? scanCustomGamesFolder(profile.customGamesFolder)
    : [];

  const seen = new Set<string>();
  const merged: PickableItem[] = [];

  for (const item of [...catalog, ...custom]) {
    if (seen.has(item.pickId)) {
      continue;
    }
    seen.add(item.pickId);
    merged.push(item);
  }

  return merged;
}

export function profileHasPickPool(profile: Profile): boolean {
  return (
    (profile.catalogGameIds?.length ?? 0) > 0 ||
    Boolean(profile.customGamesFolder?.trim())
  );
}
